### PR TITLE
Use test accounts for examples run against odsp/odspdf

### PR DESCRIPTION
**Pull request by Abe27342**
## Description

Adjusts webpack-fluid-loader's setup for odsp/odsp-df to authenticate using the same test accounts we use for running e2e tests, rather than the credentials of the person running the example. This allows considerable simplification of webpack-fluid-loader's middleware, since it no longer needs to deal with the redirects associated with more complex oauth flows. If we want to add support for that back in the future at some point, I would also personally like to see it done via a common auth library such as @azure/identity rather than our own handcrafted setup.

This change is motivated by a desire to better separate entra app registrations we use for different dev workflows against real services. As such, running examples locally is now updated to use a distinct application from our e2e tests.

### CODE REVIEW

File: examples/utils/webpack-fluid-loader/src/odspUrlResolver.ts [hunk 1]
```diff
@@ -68,16 +68,16 @@ export class OdspUrlResolver implements IUrlResolver {
 
 	public async createCreateNewRequest(fileName: string): Promise<IRequest> {
 		const filePath = "/r11s/";
-		const driveItem = await getDriveItemByRootFileName(
+		const { driveId } = await getDriveItemByRootFileName(
 			this.server,
 			"",
-			filePath,
+			"/",
```
Comment (Abe27342) [inline examples/utils/webpack-fluid-loader/src/odspUrlResolver.ts:74]: This is just a bug fix--previous version of this file made examples fail for people who didn't have an `/r11s/` folder in their drive already (which obviously wasn't true for the test accounts either).

Comment (Abe27342) [reply examples/utils/webpack-fluid-loader/src/odspUrlResolver.ts:74]: bit of a head-scratcher to find :P

File: packages/test/test-drivers/src/odspTestDriver.ts [hunk 2]
```diff
@@ -107,22 +108,32 @@ function getCredentials(
...
 		const loginAccounts =
 			odspEndpointName === "odsp"
 				? process.env.login__odsp__test__accounts
 				: process.env.login__odspdf__test__accounts;
-		assert(loginAccounts !== undefined, "Missing login__odsp/odspdf__test__accounts");
+		if (loginAccounts === undefined) {
+			// eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+			const inCi = !!process.env.TF_BUILD;
+			const odspOrOdspdf = odspEndpointName === "odsp" ? "odsp" : "odspdf";
+			throw new Error(
```
Comment (jatgarg) [inline packages/test/test-drivers/src/odspTestDriver.ts:123]: Sorry, why are we throwing error here?

Comment (Abe27342) [reply packages/test/test-drivers/src/odspTestDriver.ts:123]: I could move back to assert.fail if you want. Not super opinionated.

**Review by jatgarg — APPROVED** [review]

**Pull request merged**